packages/presentation/src/component.ts

```diff
@@ -0,0 +1,278 @@
+//
+// Copyright © 2023 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { writable } from 'svelte/store'
+
+import core, {
+  AnyAttribute,
+  Attribute,
+  Class,
+  Client,
+  Doc,
+  DocumentQuery,
+  FindOptions,
+  FindResult,
+  generateId,
+  Hierarchy,
+  Ref,
+  RefTo,
+  SortingOrder,
+  Tx,
+  TxResult
+} from '@hcengineering/core'
+import { LiveQuery } from '@hcengineering/query'
+import tracker, { Component, ComponentManager } from '@hcengineering/tracker'
+
+import { BasePresentationMiddleware, PresentationMiddleware } from './pipeline'
+
+export const componentStore = writable<ComponentManager>(new ComponentManager([]))
+
+interface ComponentSubscriber<T extends Doc = Doc> {
+  attributes: Array<Ref<AnyAttribute>>
+
+  _class: Ref<Class<T>>
+  query: DocumentQuery<T>
+  options?: FindOptions<T>
+
+  refresh: () => void
+}
+
+/**
+ * @public
+ */
+export class ComponentMiddleware extends BasePresentationMiddleware implements PresentationMiddleware {
```

Comment:
Could we unify this pipeline to support all aggregators via mixin to class, as we have for grouping?